features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/ApiConsole/ApiConsole.jsx

```diff
@@ -304,17 +304,25 @@ class ApiConsole extends React.Component {
      * @memberof ApiConsole
      */
     updateApplication() {
-        const { selectedApplication, selectedKeyType } = this.state;
+        const { selectedApplication, selectedKeyType, subscriptions } = this.state;
         const promiseApp = Application.get(selectedApplication);
         let accessToken;
+        let keyType;
+
+        if (subscriptions.find(sub => sub.applicationId === selectedApplication).status === 'PROD_ONLY_BLOCKED') {
```

Comment:
If matching subscription not found in `subscriptions ` it will return `undefined` which will throw an 

```
Uncaught TypeError: Cannot read property 'status' of undefined
```
exception in here , Shall we do a null check here ?

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/7804/commits/47c102226be6295c98512879210c7d5b920b4d98